jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/ec2/VMsScannedByTenableRule.java

```diff
@@ -19,113 +34,152 @@
+        JsonElement lastLicensedScanDate = asset.get(LAST_LICENSED_SCAN_DATE_FIELD_NAME);
+        if (tenableAssets.isEmpty() || tenableAssets.get(0).get(HAS_AGENT_FIELD_NAME).getAsBoolean() || terminatedAt != null) {
+            // FAIL: Tenable doesn't know about this asset, asset doesn't have Tenable agent installed or asset is terminated
+            Annotation annotation = getNotScannedAnnotation(ruleParam, category, entityType);
+            logger.debug("======== {} completed with annotation: {} =========", POLICY_NAME_FOR_LOGGER, annotation);
+
+            return new PolicyResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
+        } else if (!checkLastLicensedScanDate(lastLicensedScanDate, target)) {
+            // FAIL: Tenable scanned this asset more than target days ago
+            Annotation annotation = getOutdatedScanAnnotation(ruleParam, category, target, entityType);
+            logger.debug("======== {} completed with annotation:  {} =========", POLICY_NAME_FOR_LOGGER, annotation);
+
             return new PolicyResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
-          } else {
-            Optional<String> foundAny = vulnerabilityInfoList
-                .stream()
-                .map(elem -> elem.get("last_found").getAsString())
-                .filter(lastVulnScan -> {
-                  try {
-                    return PacmanUtils.calculateDuration(lastVulnScan) < Long.parseLong(target);
-                  } catch (ParseException e) {
-                    logger.error("Exception while parsing last scan date",e);
-                  }
-                  return false;
-                }).findAny();
-            if (!foundAny.isPresent()) {
-              annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
-              annotation.put(PacmanSdkConstants.DESCRIPTION, "" + entityType + " tenable not scanned since "
-                  + target + " days!!");
-              annotation.put(PacmanRuleConstants.CATEGORY, category);
-              LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
-              issue.put(PacmanRuleConstants.VIOLATION_REASON, "" + entityType + " image not scanned by tenable found");
-              issueList.add(issue);
-              annotation.put("issueDetails", issueList.toString());
-              logger.debug("========ResourceScannedByTenableRule ended with annotation {} : =========", annotation);
-              return new PolicyResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
-            }
-          }
-        } catch (Exception e) {
-          logger.error("unable to determine", e);
-          throw new RuleExecutionFailedExeption("unable to determine" + e);
         }
-      }
+
+        logger.debug("======== {} completed with no issue produced =========", POLICY_NAME_FOR_LOGGER);
```

Comment:
What's the deal with a bunch of ====?

Reply:
I think it's for visibility, will get rid of them - dont' like it as well